x-pack/plugins/ml/public/application/services/ml_api_service/inference.ts

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { useMemo } from 'react';
+import { HttpService } from '../http_service';
+import { basePath } from './index';
+import { useMlKibana } from '../../contexts/kibana';
+import { DataFrameAnalyticsConfig } from '../../data_frame_analytics/common/analytics';
+
+export interface InferenceQueryParams {
+  decompress_definition?: boolean;
+  from?: number;
+  include_model_definition?: boolean;
+  size?: number;
+  tags?: string;
+}
+
+export interface InferenceStatsQueryParams {
+  from?: number;
+  size?: number;
+}
+
+export interface InferenceConfigResponse {
+  trained_model_configs: Array<{
+    created_by: string;
+    create_time: string;
+    default_field_map: Record<string, string>;
+    estimated_heap_memory_usage_bytes: number;
+    estimated_operations: number;
+    license_level: string;
+    metadata?:
+      | {
+          analytics_config: DataFrameAnalyticsConfig;
+          input: any;
+        }
+      | Record<string, any>;
+    model_id: string;
+    tags: string;
+    version: string;
+    inference_config: any;
+  }>;
+}
+
+export type ModelConfigResponse = InferenceConfigResponse['trained_model_configs'][number];
+
+export interface IngestStats {
+  count: number;
+  time_in_millis: number;
+  current: number;
+  failed: number;
+}
+
+export interface InferenceStatsResponse {
```

Comment:
the response interface is good here, but the contents of `trained_model_stats` might be best in a common location as `trainedModelStat` or something similar.

Reply:
Refactored types in https://github.com/elastic/kibana/pull/74978/commits/d6123a31ee9f8b4e77c38694b3480d6debeab579. I had to move some DFA types to a common folder as well.